task/mgmt-gateway/src/mgs_handler.rs

```diff
@@ -152,4 +221,104 @@ impl SpHandler for MgsHandler {
             Err(ResponseError::Busy)
         }
     }
+
+    fn sys_reset_prepare(
+        &mut self,
+        _sender: SocketAddrV6,
+        _port: SpPort,
+    ) -> Result<(), ResponseError> {
+        ringbuf_entry!(Log::MgsMessage(MgsMessage::SysResetPrepare));
+        self.reset_requested = true;
+        Ok(())
+    }
+
+    fn sys_reset_trigger(
+        &mut self,
+        _sender: SocketAddrV6,
+        _port: SpPort,
+    ) -> Result<Infallible, ResponseError> {
+        if !self.reset_requested {
+            return Err(ResponseError::SysResetTriggerWithoutPrepare);
+        }
+
+        let jefe = task_jefe_api::Jefe::from(crate::JEFE.get_task_id());
+        jefe.request_reset();
```

Comment:
Not thrilled that anyone who can deliver a UDP packet to an SP can cause the SP to reset.  If nothing else, add a TODO for "insert auth checks here"